api/src/main/java/ai/djl/ndarray/NDScope.java

```diff
@@ -28,7 +28,7 @@ public class NDScope implements AutoCloseable {
     private static final ThreadLocal<Deque<NDScope>> SCOPE_STACK =
             ThreadLocal.withInitial(ArrayDeque::new);
 
-    private List<NDArray> resources;
+    private List<NDArrayWrapper> resources;
```

Comment:
We can use IdentityHashMap, we don't need introduce a new NDArrayWrapper class, and we don't need do a linear search.
```suggestion
    private IdentityHashMap<NDArray, NDArray> resources;
```